src/Type/Generic/TemplateKeyOfType.php

```diff
@@ -0,0 +1,54 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Generic;
+
+use PHPStan\Type\KeyOfType;
+use PHPStan\Type\Traits\UndecidedComparisonCompoundTypeTrait;
+use PHPStan\Type\Type;
+
+/** @api */
+final class TemplateKeyOfType extends KeyOfType implements TemplateType
```

Comment:
It's weird to add support for key-of in generics but not for value-of at the same time :)

Reply:
I did have it implemented, but I wasn't able to think of a good use-case for it.

Reply:
I don't either but it looks like an ommission not to have it :)